capture_hdr_loop.py

```diff
@@ -1,68 +1,65 @@
+to grab good data on most scenes, including shiny for a medium camera operating
+at around 0.7m-1.2m
+
 """
 
 import datetime
+from pathlib import Path
+import yaml
 import zivid
 
 
+def _get_settings(settings: zivid.Settings, path: Path):
+    settings_from_yaml = yaml.load(open(path, "rb").read(), Loader=yaml.Loader)["Settings"]
+    settings.brightness = settings_from_yaml["Brightness"]
+    settings.exposure_time = datetime.timedelta(
+        microseconds=settings_from_yaml["ExposureTime"]
+    )
+    settings.gain = settings_from_yaml["Gain"]
+    settings.iris = settings_from_yaml["Iris"]
+    filters = settings_from_yaml["Filters"]
+    settings.filters.contrast.enabled = filters["Contrast"]["Enabled"]
+    settings.filters.contrast.threshold = filters["Contrast"]["Threshold"]
+    settings.filters.gaussian.enabled = filters["Gaussian"]["Enabled"]
+    settings.filters.gaussian.sigma = filters["Gaussian"]["Sigma"]
+    settings.filters.outlier.enabled = filters["Outlier"]["Enabled"]
+    settings.filters.outlier.threshold = filters["Outlier"]["Threshold"]
+    settings.filters.reflection.enabled = filters["Reflection"]["Enabled"]
+    settings.filters.saturated.enabled = filters["Saturated"]["Enabled"]
+
+
 def _main():
     app = zivid.Application()
 
     print("Connecting to the camera")
-    camera = app.connect_camera()
-
-    print("Configuring the camera settings")
-    settings_collection = [camera.settings for _ in range(3)]
-
-    settings_collection[0].brightness = 1.8
-    settings_collection[0].gain = 1
-    settings_collection[0].filters.gaussian.enabled = True
-    settings_collection[0].filters.gaussian.sigma = 1.5
-    settings_collection[0].filters.reflection.enabled = True
-
-    settings_collection[1].brightness = 1.8
-    settings_collection[1].gain = 1
-    settings_collection[1].filters.gaussian.enabled = True
-    settings_collection[1].filters.gaussian.sigma = 1.5
-    settings_collection[1].filters.reflection.enabled = True
-
-    settings_collection[2].brightness = 1.8
-    settings_collection[2].gain = 1
-    settings_collection[2].filters.gaussian.enabled = True
-    settings_collection[2].filters.gaussian.sigma = 1.5
-    settings_collection[2].filters.reflection.enabled = True
-
-    exposure_time_frame_1 = [
-        datetime.timedelta(microseconds=10000),
-        datetime.timedelta(microseconds=90000),
-        datetime.timedelta(microseconds=40000),
-    ]
-    exposure_time_frame_2 = [
-        datetime.timedelta(microseconds=40000),
-        datetime.timedelta(microseconds=10000),
-        datetime.timedelta(microseconds=90000),
-    ]
-    exposure_time_frame_3 = [
-        datetime.timedelta(microseconds=90000),
-        datetime.timedelta(microseconds=40000),
-        datetime.timedelta(microseconds=10000),
-    ]
-    iris_frames = [10, 20, 30]
+    # camera = app.connect_camera()
+    camera = app.create_file_camera("C:\\Program Files\\Zivid\\data\\MiscObjects.zdf")
```

Comment:
I guess you want to delete this one and uncomment line 39.

Reply:
Does this work btw? I think maybe we should keep it but comment out. What do you think?